dali/pipeline/data/tensor_vector.h

```diff
@@ -254,8 +254,8 @@ class TensorVector {
     if (!IsValidType(tl_->type())) return;
     if (!tl_->raw_data()) return;
 
+    views_count_ = tensors_.size();
     for (size_t i = 0; i < tensors_.size(); i++) {
```

Comment:
```suggestion
    for (size_t i = 0; i < views_count_; i++) {
```

Reply:
This is an atomic counter, I don't think we want to use it here.

Reply:
That's a very bad idea, actually. `views_count_` is an atomic variable and, as such, is implicitly volatile and will force global bus transactions in every iteration.